### PR TITLE
Add work remotely option in Create Post and Filter Page

### LINKED ISSUE #2180: Add Remote Work Feature on Create Post and Filter Pages
Add a question for user in ‘Create Post’ if they are available to work/volunteer remotely. Refer below Notion link for more details:
[Feature Remote Work](https://www.notion.so/fightpandemics/Post-should-be-created-and-filtered-as-per-Remote-option-d1767e7eb31f49b59855c90c0573d206)

In Addition to above notion requirements;

1. Hide the 'Remote Work' label from 'Create Post' modal
2. For old/existing post-> set the value of 'Work remotely' question as: 'Yes' if 'Remote work' label is selected for any post else set NO
3. For new post-> if user select 'Yes' to question 'can work remotely' then the post should have 'Remote Work' label, just like we have now.

### PR DESCRIPTION
**Pull request by aditilamkhede**
### What does this pull request aim to achieve? 💯
Add work remotely option in Create Post and in Filter on feed Page.

_Please be concise and link any related issue(s):_
adding a feature resolves issue #2180 

- [x] I have checked that no one else is working on similar changes.
- [x] I have read the latest [**README**](README.md) and understand the development workflow.
- [x] The name of this branch is: **`feature/<branch_name>`**, or **`hotfix/<branch_name>`** if this is a hotfix, where `<branch_name>` briefly describes the issue(s) resolved and is prefixed with the issue number(s) (e.g. `1127-1130-update-git-branching-model`). The branch is created in a cloned version of this repo, **not a forked repo**.
- [x] This branch is rebased/merged with the **latest staging** (or the **latest production** if it is a hotfix).
- [x] There are no merge conflicts.
- [x] There are no console warnings and errors.
- [x] On the right hand side of this PR, I have linked the appropriate issue(s) under "Issues" and added a project progress under "Projects".
- [x] The title describes the issue(s) being addressed using format: **`<issue_numbers> - <brief_issue_description>`** (e.g. `#1127/#1130 - Update Git Branching Model`)

### CODE REVIEW

**Comment by aditilamkhede** [comment]
@rabbi08 here is the build URL: https://featureadd-remote-createpostandfilter-pa.fightpandemics.xyz/

**Comment by rabbi08** [comment]
thanks @aditilamkhede 

below are few observations for this feature:

1. Please update label of 'Remote' as 'Remote Work' on Filters page.

2. The size, color and font of 'Remote option's "TEXT and ICON" should match with 'Share my location' option
Also great if we can align this option with 'Share my location' Please see below screenshot from desktop:
image

3. Search is performed when user click on 'Remote' option only. User should be able to select the checkbox and click on 'Apply Filters' to see the search results, just like for other filter options. This is to keep the search filters in sync, so that user can combine more than one filters.

4. After 'Remote' filter is applied and user is directed to help board, the 'applied filter option-remote' should be displayed to indicate the user that remote filter is in place, similar to other filter options. 

Please see below video for issue# 2 & 3:

https://user-images.githubusercontent.com/67168400/106177295-51b05080-6166-11eb-8885-81b86148714e.movPlease refer above video.

5. Filter: Existing post having tag as 'Remote Work' are NOT displayed in the search. Please see below video:

https://user-images.githubusercontent.com/67168400/106180275-17e14900-616a-11eb-9503-a2d20c2ee628.mov

6. Filtered Results: the post having no 'Remote work' tag is displayed. Please see below video:

https://user-images.githubusercontent.com/67168400/106181073-22e8a900-616b-11eb-9429-7e29e927bc2f.mov

**Comment by aditilamkhede** [comment]
@rabbi08 I just have to work on 4th from above and then I will be done. 5 and 6are related to update data for old records. I will create script for old records and send it to @ddveloper.

**Comment by aditilamkhede** [comment]
@rabbi08 Ready for testing. 
Here is the build url: https://featureadd-remote-createpostandfilter-pa.fightpandemics.xyz/feed

**Review by rabbi08 — CHANGES_REQUESTED** [review]
Thanks @aditilamkhede 
Below are few initial observations, while testing this feature:
1. In Create Post window, please add a sentence below Question and align the dropdown with the visibility dropdown:
The Post can be fulfilled <dropdown>.

2. On HelpBoard -> Filter, Please update the filter heading 'Remote Work' as 'Post fulfillment' 

3. For Old Post having <Remote> tag 
    a. Are displayed as 'Both i.e. Remote and In- Person'  they should be 'Remote' not 'Both'
    b. and 'Remote' TAG is displayed on the post, we should NOT display the Remote tag on the post.
Please see below screenshot.
    
image

**Comment by aditilamkhede** [comment]
@rabbi08 in addition to the 3rd observation from above,I believe we should remove "Remote Work" tag from "Types" filters as well on HelpBoard

**Comment by rabbi08** [comment]
> @rabbi08 in addition to the 3rd observation from above,I believe we should remove "Remote Work" tag from "Types" filters as well on HelpBoard

Yes plz
great catch @aditilamkhede 
Thanks

**Comment by aditilamkhede** [comment]
@rabbi08 ready for review.

**Comment by SinghRomika** [comment]
@aditilamkhede The question in the Create Post pop up modal window on the build  does not match the Feature description given on [Notion](https://www.notion.so/fightpandemics/Post-should-be-created-and-filtered-as-per-Remote-option-d1767e7eb31f49b59855c90c0573d206).
Capture1

Create Post pop up modal window on the build.
Capture2

**Comment by aditilamkhede** [comment]
> @aditilamkhede The question in the Create Post pop up modal window on the build does not match the Feature description given on [Notion](https://www.notion.so/fightpandemics/Post-should-be-created-and-filtered-as-per-Remote-option-d1767e7eb31f49b59855c90c0573d206).
> Capture1
> 
> Create Post pop up modal window on the build.
> Capture2

@SinghRomika you are right, but notion needs to be updated. Notion is old. In current scenario there will be 3 options as you see currently.

**Comment by SinghRomika** [comment]
@aditilamkhede  while reviewing the PR #2165 (new post fulfillment option in 'create post' and 'filter' page).
When i clicked the search my Location( Under Location). The location captured by the application was different than my original location.
(Location captured by the application)
Filter Location
(The difference between the Actual location vs what the application shows)
Filter Location3_LI

**Comment by aditilamkhede** [comment]
> @aditilamkhede while reviewing the PR #2165 (new post fulfillment option in 'create post' and 'filter' page).
> When i clicked the search my Location( Under Location). The location captured by the application was different than my original location.
> (Location captured by the application)
> Filter Location
> (The difference between the Actual location vs what the application shows)
> Filter Location3_LI

Hi @SinghRomika not part of this PR. So, may be you want to file a bug. Also check the same on Staging before filing a bug.
@rabbi08 ?

**Comment by SinghRomika** [comment]
@aditilamkhede I added this comment by mistake, please ignore it. I will check and review the Staging also.

**Comment by SinghRomika** [comment]
@aditilamkhede Reviewed PR #2165 for now there seems to be no issue in the PR #2165 for desktop . Thanks!

**Comment by rabbi08** [comment]
Thanks @aditilamkhede 
The features looks great, the functionality is working as expected in Create post and 'filter' page.
I have an observation on mobile view:
- When user EDIT the post and have remote options value as both i.e. 'Remote & In-Person' the UI get distorted for the post i.e. the ellipses gets overlapped and the 'edited' text moved on the user profile icon. Please see below GIF and screenshot.

IMG_2C0096AC6F77-1
ezgif com-video-to-gif-10

**Comment by aditilamkhede** [comment]
> Thanks @aditilamkhede
> The features looks great, the functionality is working as expected in Create post and 'filter' page.
> I have an observation on mobile view:
> 
> * When user EDIT the post and have remote options value as both i.e. 'Remote & In-Person' the UI get distorted for the post i.e. the ellipses gets overlapped and the 'edited' text moved on the user profile icon. Please see below GIF and screenshot.
> 
> IMG_2C0096AC6F77-1
> ezgif com-video-to-gif-10

@rabbi08 I will see what I can do. If there isn't any space available are we ok to trim some text or any other suggestions like decreasing font size?

**Comment by rabbi08** [comment]
> > Thanks @aditilamkhede
> > The features looks great, the functionality is working as expected in Create post and 'filter' page.
> > I have an observation on mobile view:
> > 
> > * When user EDIT the post and have remote options value as both i.e. 'Remote & In-Person' the UI get distorted for the post i.e. the ellipses gets overlapped and the 'edited' text moved on the user profile icon. Please see below GIF and screenshot.
> > 
> > IMG_2C0096AC6F77-1
> > ezgif com-video-to-gif-10
> 
> @rabbi08 I will see what I can do. If there isn't any space available are we ok to trim some text or any other suggestions like decreasing font size?

Yes @aditilamkhede  we can trim for small size screens, if that work
Trim date can be like -
"Remote/In-pers" if that works

Thanks

File: client/src/components/CreatePost/Form/SecondSection.js [hunk 1]
```diff
@@ -4,7 +4,10 @@ import Head from "./Head";
 import { Section } from "components/CreatePost/StyledModal";
 import filterOptions from "assets/data/filterOptions";
 
-const types = Object.values(filterOptions)[2].options;
+let types = Object.values(filterOptions)[2].options;
+types = types.filter((val) => {
+  return val.value != "Remote Work";
```
Comment (ddveloper) [inline client/src/components/CreatePost/Form/SecondSection.js]: may be use !== instead ?

File: client/src/components/Feed/FilterAccordion.js [hunk 2]
```diff
@@ -81,6 +137,8 @@ const FilterAccord = ({ gtmPrefix, locationOnly }) => {
     dispatchAction(SET_VALUE, "activePanel", activePanelKey);
   };
 
+  console.log("filters", filters, gtmPrefix);
```
Comment (ddveloper) [inline client/src/components/Feed/FilterAccordion.js]: remove console.log() ?

File: client/src/components/Feed/FilterAccordion.js [hunk 3]
```diff
@@ -114,22 +172,25 @@ const FilterAccord = ({ gtmPrefix, locationOnly }) => {
...
-                <ButtonTag
-                  id={gtmPrefix + filterOps(filter.label, idx)}
-                  key={idx}
-                  onClick={handleOption(filter.label, value)}
-                  className={tagClassName}
-                >
-                  {t(text)}
-                </ButtonTag>
-              );
+              console.log("idx", idx);
```
Comment (ddveloper) [inline client/src/components/Feed/FilterAccordion.js]: remove console.log() ?

File: client/src/components/Feed/Post.js [hunk 4]
```diff
@@ -604,6 +608,17 @@ const Post = ({
             )}
             <div className="pre-header post-page">
               <span>{t(`feed.${objective}`)}&nbsp;&nbsp;•</span>
+              <span>
+                &nbsp;&nbsp;
```
Comment (ddveloper) [inline client/src/components/Feed/Post.js:621]: try <Space/> or other antd components? `&nbsp;` might not ideal

File: client/src/components/Feed/Post.js [hunk 5]
```diff
@@ -741,6 +756,17 @@ const Post = ({
               )}
               <div className="pre-header">
                 <span>{t(`feed.${objective}`)}&nbsp;&nbsp;•</span>
+                <span>
+                  &nbsp;&nbsp;
+                  {t(
+                    workMode && workMode !== "both"
```
Comment (ddveloper) [inline client/src/components/Feed/Post.js]: This logic seems repeated several times, maybe extract it out

File: client/src/pages/Feed.js [hunk 6]
```diff
@@ -310,13 +316,38 @@ const Feed = (props) => {
...
     setQueryKeysValue(history, { location: null });
+
     setTimeout(() => {
       dispatchAction(SET_VALUE, "activePanel", null);
     }, 500);
     // perform soft refetch to only close filter modal etc.. but not actually refetch posts
     refetchPosts(null, null, true);
   };
 
+  /* //Aditi Do I need this */
```
Comment (ddveloper) [inline client/src/pages/Feed.js]: maybe remove the commented code?

File: backend/lib/endpoints/posts.js [hunk 7]
```diff
@@ -113,6 +112,22 @@ async function routes(app) {
         filters.push({ types: { $in: type } });
       }
 
+      //workMode filter
+      if (workMode){
+        const workModes = workMode.map(mode => mode.toLowerCase())
+        if(workModes.includes('both')){
+          console.log("In both")
```
Comment (ddveloper) [inline backend/lib/endpoints/posts.js]: remove the console.log() ?

File: backend/lib/endpoints/posts.js [hunk 8]
```diff
@@ -113,6 +112,22 @@ async function routes(app) {
...
+          filters.push({ $or: [ { $and: [ { workMode: null }, {types: { $nin : ['Remote Work'] } } ] }, {workMode: { $in: workModes } }] });
+        }
+        else if(workModes.includes('remote')) {
+          console.log("In remote")
```
Comment (ddveloper) [inline backend/lib/endpoints/posts.js]: remove the console.log() ?

File: backend/lib/endpoints/posts.js [hunk 9]
```diff
@@ -113,6 +112,22 @@ async function routes(app) {
...
+          filters.push({ $or: [ { $and: [ { workMode: null }, {types: { $in : ['Remote Work'] } } ] }, {workMode: { $in: workModes } }] });
```
Comment (ddveloper) [inline backend/lib/endpoints/posts.js:122]: This query seems complex, might be a better (simpler) way to achieve the goal ❔ 
it seems you are not using formatting tool Aditi, the format changes when I run prettier in VS code

File: client/src/components/CreatePost/Form/SecondSection.js (hunk 1 above)
Comment (aditilamkhede) [reply client/src/components/CreatePost/Form/SecondSection.js]: Done

File: client/src/components/Feed/FilterAccordion.js (hunk 2 above)
Comment (aditilamkhede) [reply client/src/components/Feed/FilterAccordion.js]: Done

File: client/src/components/Feed/FilterAccordion.js (hunk 3 above)
Comment (aditilamkhede) [reply client/src/components/Feed/FilterAccordion.js]: Done

File: client/src/components/Feed/Post.js (hunk 4 above)
Comment (aditilamkhede) [reply client/src/components/Feed/Post.js:621]: tried using WhiteSpace - puts it into next line, antd Space doesn't help

Comment (ddveloper) [reply client/src/components/Feed/Post.js:612]: the [antd <Space>](https://ant.design/components/space/) might be useful here

File: client/src/components/Feed/Post.js (hunk 5 above)
Comment (aditilamkhede) [reply client/src/components/Feed/Post.js]: Done

File: client/src/pages/Feed.js (hunk 6 above)
Comment (aditilamkhede) [reply client/src/pages/Feed.js]: Done

File: backend/lib/endpoints/posts.js (hunk 7 above)
Comment (aditilamkhede) [reply backend/lib/endpoints/posts.js]: Done

File: backend/lib/endpoints/posts.js (hunk 8 above)
Comment (aditilamkhede) [reply backend/lib/endpoints/posts.js]: done

File: backend/lib/endpoints/posts.js (hunk 9 above)
Comment (aditilamkhede) [reply backend/lib/endpoints/posts.js:122]: @ddveloper this one is basically adds logic to filter old records which don't have workMode fields but can have remote work tag in types feild.

**Review by rabbi08 — APPROVED** [review]
Thanks @aditilamkhede 
Verified the fix on iPhone 11 Pro Max, this is working fine. Pleas see below screenshot. 
image
LGTM

**Review by ddveloper — APPROVED** [review]
Thanks @aditilamkhede , LGTM! 👍

**Pull request merged**